src/modules/modal/components/modal-network-connect/modal-network-connect.jsx

```diff
@@ -0,0 +1,215 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import classNames from 'classnames'
+
+import { AugurLoadingLogo, ExclamationCircle as InputErrorIcon } from 'modules/common/components/icons'
+import Input from 'modules/common/components/input/input'
+
+import Styles from 'modules/modal/components/modal-network-connect/modal-network-connect.styles'
+
+function calculateConnectionErrors(err, res) {
+  const errors = []
+
+  if (err || (!!res && !res.ethereumNode && !res.augurNode)) errors.push('There was an issue connecting to the nodes, please try again.')
+  if (!!res && !res.ethereumNode && !err && res.augurNode) errors.push('Failed to connect to the Ethereum Node.')
+  if (!!res && !res.augurNode && !err && res.ethereumNode) errors.push('Failed to connect to the Augur Node.')
+
+  return errors
+}
+
+export default class ModalNetworkConnect extends Component {
+  static propTypes = {
+    modal: PropTypes.shape({
+      isInitialConnection: PropTypes.bool,
+    }),
+    env: PropTypes.object,
+    connection: PropTypes.object,
+    submitForm: PropTypes.func.isRequired,
+    closeModal: PropTypes.func.isRequired,
+    updateEnv: PropTypes.func.isRequired,
+  }
+
+  constructor(props) {
+    super(props)
+    // prioritize ethereumNode connections
+    let ethereumNode = ''
+    if (props.env['ethereum-node']) {
+      if (props.env['ethereum-node'].ipc) {
+        ethereumNode = props.env['ethereum-node'].ipc
+      } else if (props.env['ethereum-node'].ws) {
+        ethereumNode = props.env['ethereum-node'].ws
+      } else if (props.env['ethereum-node'].http) {
+        ethereumNode = props.env['ethereum-node'].http
+      }
+    }
+
+    this.state = {
+      augurNode: props.env['augur-node'] || '',
+      ethereumNode,
+      isAttemptingConnection: false,
+      isWeb3Available: (!!window && !!window.web3),
```

Comment:
Should we be polling for this?

Reply:
I don't think that's needed, basically I only need to care if it's there on this form so we can hide the input for ethereumNode because you are already connected through metamask. I can't think of a reason we would poll for it over just checking if it's there or not when we go to construct this component.